### ISSUE TITLE
Wrong error with function overloads 

### ISSUE THREAD

**Issue by paleo** [Microsoft/TypeScript#14042]
**TypeScript Version:**  2.1.6

**Code**

```ts
type Person = {
    name: string
}
type Task = {
    label: string
}

function exec(cmd: "create", p: Person)
function exec(cmd: "update", t: Task)

function exec(cmd: string, pOrT) {
    // Implementation here
}

exec("create", { falseMember: "abc" }) // KO: Error on parameter: "create"
exec("update", { name: "abc" }) // OK: Error on the last parameter
```

**Expected behavior:**

In the following code:

```ts
exec("create", { falseMember: "abc" })
```

... the error should be:

> Argument of type '{falseMember: string}' is not assignable to parameter of type 'Person'

I see two ways to implement it:

1. When all parameters match to a valid type except one, the error should be on the invalid parameter;
2. Or, the compiler should prefer to show an error on the last parameter, not the first.

**Actual behavior:**

In the following code:

```ts
exec("create", { falseMember: "abc" })
```

... the error is:

> Argument of type '"create"' is not assignable to parameter of type '"update"'

**Comment by RyanCavanaugh** [comment]
See #10523
